From 2da919ff8eefcf4e4313e840e31844fa2468b216 Mon Sep 17 00:00:00 2001
From: Reza Arbab <arbab@linux.ibm.com>
Date: Fri, 13 Sep 2019 15:15:30 -0500
Subject: [PATCH 1/3] Fix NV_ON_EACH_CPU()

Since commit caa759323c73 ("smp: Remove smp_call_function() and
on_each_cpu() return values"), on_each_cpu() has no return value.

kernel/common/inc/nv-linux.h:733:21: error: void value not ignored as it ought to be
         int __ret = on_each_cpu(func, info, 1);        \
                     ^

Fix the ensuing compile error.
---
 kernel/common/inc/nv-linux.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/common/inc/nv-linux.h b/kernel/common/inc/nv-linux.h
index da61baab0250..a74d59910ebe 100644
--- a/kernel/common/inc/nv-linux.h
+++ b/kernel/common/inc/nv-linux.h
@@ -730,8 +730,8 @@ static inline void nv_vunmap(NvUPtr vaddr, NvU32 page_count)
 #elif (NV_ON_EACH_CPU_ARGUMENT_COUNT == 3)
 #define NV_ON_EACH_CPU(func, info)                     \
     ({                                                 \
-        int __ret = on_each_cpu(func, info, 1);        \
-        __ret;                                         \
+        on_each_cpu(func, info, 1);                    \
+        0;                                             \
      })
 #else
 #error "NV_ON_EACH_CPU_ARGUMENT_COUNT value unrecognized!"
-- 
1.8.3.1

